sql/parser/eval.go

```diff
@@ -471,6 +658,12 @@ var binOps = map[binArgs]func(Datum, Datum) (Datum, error){
 	binArgs{Div, floatType, floatType}: func(left Datum, right Datum) (Datum, error) {
 		return left.(DFloat) / right.(DFloat), nil
 	},
+	binArgs{Div, intervalType, intType}: func(left Datum, right Datum) (Datum, error) {
+		if right.(DInt) == 0 {
+			return DInterval{}, util.Error("division by zero error")
```

Comment:
Ah, I forgot that we converted ints to floats, so this is the only place that we actually do integer division. Returning the error is fine, then (but maybe add a comment).